### PR TITLE
Added windows support for the plugin

### PR DESCRIPTION
**Pull request by saleem-latif**

### CODE REVIEW

File: src/main/java/org/intellij/sdk/codesync/ProjectOpenCloseListener.java [hunk 1]
```diff
@@ -50,12 +50,24 @@ public void projectOpened(@NotNull Project project) {
       public void after(@NotNull List<? extends VFileEvent> events) {
         String repoPath = project.getBasePath();
 
+        if (Utils.isWindows()){
+          // For some reason people at intellij thought it would be a good idea to confuse users by using
```
Comment (Basit-qc) [inline src/main/java/org/intellij/sdk/codesync/ProjectOpenCloseListener.java:54]: really? :laughing:

File: src/main/java/org/intellij/sdk/codesync/Utils.java [hunk 2]
```diff
@@ -8,41 +8,55 @@
...
+import java.util.regex.Pattern;
 import java.util.stream.Stream;
 
 import static org.intellij.sdk.codesync.Constants.*;
 
 public class Utils {
+    private static String OS = System.getProperty("os.name").toLowerCase();
+
+    public static boolean isWindows() {
+        return OS.contains("win");
```
Comment (Basit-qc) [inline src/main/java/org/intellij/sdk/codesync/Utils.java:33]: nice :+1:

File: src/main/java/org/intellij/sdk/codesync/repoManagers/DeletedRepoManager.java [hunk 3]
```diff
@@ -60,9 +70,15 @@ public void copyFiles(String[] filePaths) {
     */
     public void copyFiles(String[] filePaths, String projectRepoPath) {
         for (String filePath: filePaths) {
+            String originalFilePath = filePath;
+
+            if (Utils.isWindows()) {
```
Comment (Basit-qc) [inline src/main/java/org/intellij/sdk/codesync/repoManagers/DeletedRepoManager.java:75]: I think we need this only when copying within system directories. i.e. from .shadow to .delete. 
When copying from actual repo, we need `:` in the source path.
btw, I might have missed, where are we using `filePath` variable?

File: src/main/java/org/intellij/sdk/codesync/ProjectOpenCloseListener.java (hunk 1 above)
Comment (saleem-latif) [reply src/main/java/org/intellij/sdk/codesync/ProjectOpenCloseListener.java:54]: ref: https://youtrack.jetbrains.com/issue/IDEA-228825#focus=streamItem-27-4231153.0-0

**Pull request merged**